### ISSUE TITLE
ld.so.preload issue when running x32 apps on x64 host

### ISSUE THREAD

**Issue by djfd** [netblue30/firejail#971]
Hi,

currently this simply does not work. E.g.
```
Child process initialized
ERROR: ld.so: object '/usr/lib/firejail/libconnect.so' from /etc/ld.so.preload cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
monitoring pid 3
```

Could you please change build script to produce required x32 libraries on x64 hosts and also add a piece of logic to startup routine to supply right libraries when starting x32 apps...

thanks

**Comment by netblue30** [comment]
You are right, currently the libraries are hardcocde to the original build (64bit in this case). I'll set is as an enhancement, thanks.

**Comment by djfd** [comment]
that will not be easy, it seems... there are compiled apps, but also there are various shell scripts launchers and it is impossible to know what binaries they will load...

possible need to supply both x32 and x64 libraries in x64 environments...

**Comment by netblue30** [comment]
I'll try to see how far I can go on this one. Building the libraries is not such a problem. I don't know in this moment how to detect what library should be loaded. I am looking at Wine, it starts up as a 64bit executable and then it switches to 32bit.

**Comment by djfd** [comment]
it seems the only way to supply both kinds simultaneously?

eg.one can run
```firejail... --shell=none bash -c "xterm-x64 & ; untrusted-x32"``` ))) I am doing so to check what is going on ))

**Comment by rusty-snake** [comment]
still an issue?
